src/machinelist.cpp

```diff
@@ -955,259 +947,264 @@ void MachineList::parseMachineDetail(QTreeWidgetItem *item)
-			attributes.clear();
-			attributes << "tag" << "mandatory" << "interface";
-			descriptions.clear();
-			descriptions << tr("Tag") << tr("Mandatory") << tr("Interface");
-			insertAttributeItems(childItem, element, attributes, descriptions, false);
-			gamePos++;
-			while ( xmlLines.at(gamePos).contains("<instance ") ) {
-				QString subElement(xmlLines.at(gamePos).simplified());
-				QTreeWidgetItem *secondChildItem = new QTreeWidgetItem(childItem);
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_MACHINE, tr("Instance"));
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_ICON, value(subElement, "name", false));
-				attributes.clear();
-				attributes << "briefname";
-				descriptions.clear();
-				descriptions << tr("Brief name");
-				insertAttributeItems(secondChildItem, element, attributes, descriptions, false);
-				gamePos++;
-			}
-			while ( xmlLines.at(gamePos).contains("<extension ") ) {
-				QString subElement(xmlLines.at(gamePos).simplified());
-				QTreeWidgetItem *secondChildItem = new QTreeWidgetItem(childItem);
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_MACHINE, tr("Extension"));
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_ICON, value(subElement, "name", false));
-				gamePos++;
-			}
-		}
-		if ( element.contains("<ramoption") ) {
-			childItem = new QTreeWidgetItem();
-			childItem->setText(QMC2_MACHINELIST_COLUMN_MACHINE, tr("RAM options"));
-			while ( xmlLines.at(gamePos).contains("<ramoption") ) {
-				QString subElement(xmlLines.at(gamePos).simplified());
-				QTreeWidgetItem *secondChildItem = new QTreeWidgetItem(childItem);
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_MACHINE, tr("Option"));
-				int fromIndex = subElement.indexOf('>') + 1;
-				int toIndex = subElement.indexOf('<', fromIndex);
-				QString ramOptionValue(subElement.mid(fromIndex, toIndex - fromIndex));
-				secondChildItem->setText(QMC2_MACHINELIST_COLUMN_ICON, ramOptionValue);
-				attributes.clear();
-				attributes << "default";
-				descriptions.clear();
-				descriptions << tr("Default");
-				insertAttributeItems(secondChildItem, subElement, attributes, descriptions, false);
-				gamePos++;
+
+	if ( xmlMachineEntry.readNextStartElement() ) {
+		if ( xmlMachineEntry.name() == "machine" ) {
+			insertAttributeItems(&itemList, &xmlMachineEntry, attributes, descriptions, true);
+			while ( xmlMachineEntry.readNextStartElement() ) {
+				if ( xmlMachineEntry.name() == "year" ) {
+					content = xmlMachineEntry.readElementText();
+					childItem = new QTreeWidgetItem();
+					childItem->setText(QMC2_MACHINELIST_COLUMN_MACHINE, tr("Year"));
+					childItem->setText(QMC2_MACHINELIST_COLUMN_ICON, content);
+				} else if ( xmlMachineEntry.name() =="manufacturer" ) {
+					content = xmlMachineEntry.readElementText();
+					content.replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
```

Comment:
More XML character entity decoding you shouldn’t be doing any more.

Reply:
Done